src/Symfony/Component/Serializer/Mapping/Loader/YamlFileLoader.php

```diff
@@ -87,11 +87,29 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata)
                 }
 
                 if (isset($data['serialized_name'])) {
-                    if (!\is_string($data['serialized_name']) || empty($data['serialized_name'])) {
-                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                    if (!\is_string($data['serialized_name']) && !\is_array($data['serialized_name'])) {
+                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string or an array of serialized name/groups in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
                     }
 
-                    $attributeMetadata->setSerializedName($data['serialized_name']);
+                    if (\is_string($data['serialized_name'])) {
+                        if (empty($data['serialized_name'])) {
+                            throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                        }
+                        $attributeMetadata->addSerializedName($data['serialized_name']);
+                    }
+
+                    if (\is_array($data['serialized_name'])) {
```

Comment:
This should be `elseif` or in previous `if` should be `continue` cause that value cannot be both, an string & array.

Reply:
true
done